Android App/app/src/main/java/org/cfp/citizenconnect/Model/Layout.java

```diff
@@ -48,14 +71,37 @@ public void setName(String name) {
         this.name = name;
     }
 
-    public static void getDataSetLayout(DatabaseReference databaseReference,CustomCallBack.Listener<ArrayList<Layout>> mList, CustomCallBack.ErrorListener<DatabaseError> mErr) {
+    public static void getLayout(String type,DatabaseReference databaseReference, CustomCallBack.Listener<ArrayList<Layout>> mList, CustomCallBack.ErrorListener<DatabaseError> mErr) {
         ArrayList<Layout> list = new ArrayList<>();
-        getAFireBaseData(databaseReference, response -> {
-            for (DataSnapshot _child : response.getChildren()) {
-                Layout mLayout = _child.getValue(Layout.class);
-                list.add(mLayout);
+        RealmResults<Layout> results = realm.where(Layout.class).equalTo("type",type).findAll();
+        if(results.size()>0){
+            for(Layout layout :results){
+                list.add(layout);
             }
```

Comment:
no need to loop through all items as this is not an optimized way. You can replace it with just `list.addAll(results)` and it will serve as the same purpose.

Reply:
agreed